### PR TITLE
185505938 link prevention assessment report

### PR DESCRIPTION
**Pull request by ZachCMP**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Adds logic to fetch the LINK Prevention Assessment Report and return it to the client

Depend on frontend PR: https://github.com/greenriver/hmis-frontend/pull/576

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis/app/controllers/ac_hmis/reports_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module AcHmis
+  class ReportsController < Hmis::BaseController
+    def prevention_assessment_report
```
Comment (gigxz) [inline drivers/hmis/app/controllers/ac_hmis/reports_controller.rb:9]: We need some additional checks here to ensure that the calling user has access to view the Referral in question. Can you try this out (untested)

```
      referral_id = params[:referral_id]
      referral = HmisExternalApis::AcHmis::Referral.find_by(identifier: referral_id)
      raise HmisErrors::ApiError, "Referral not found (ID: #{referral_id})" unless referral.present?

      # Find the project that this referral is for, and ensure that user has permission to view postings at it
      project_referred_to = referral.postings&.first&.project
      raise HmisErrors::ApiError, "Project not found for referral (ID: #{referral_id})" unless project_referred_to
      raise HmisErrors::ApiError, 'Access denied' unless current_user.can_manage_incoming_referrals_for?(project_referred_to)

```

Comment (ZachCMP) [reply drivers/hmis/app/controllers/ac_hmis/reports_controller.rb:9]: @gigxz Just put that in. I used your logic with a couple modifications.

File: drivers/hmis/app/models/ac_hmis/report_api.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#   "id"=>6,
+#   "type"=>"GrdaWarehouse::RemoteCredentials::Oauth",
+#   "active"=>true,
+#   "username"=>"",
+#   "encrypted_password"=>"",
+#   "encrypted_password_iv"=>nil,
+#   "region"=>nil,
+#   "bucket"=>"API_TEST",
+#   "path"=>"https://alleghenycounty-demo.oktapreview.com/oauth2/TOKEN/v1/token",
+#   "endpoint"=>"https://dhsaz-apim-api-uat.azure-api.net/green-river-api/api",
```
Comment (gigxz) [inline drivers/hmis/app/models/ac_hmis/report_api.rb]:
```suggestion
#   "endpoint"=>"https://BASE/green-river-api/api",
```

File: drivers/hmis/app/models/ac_hmis/report_api.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+# creds.attributes = {
+#   "id"=>6,
+#   "type"=>"GrdaWarehouse::RemoteCredentials::Oauth",
+#   "active"=>true,
+#   "username"=>"",
+#   "encrypted_password"=>"",
+#   "encrypted_password_iv"=>nil,
+#   "region"=>nil,
+#   "bucket"=>"API_TEST",
+#   "path"=>"https://alleghenycounty-demo.oktapreview.com/oauth2/TOKEN/v1/token",
```
Comment (gigxz) [inline drivers/hmis/app/models/ac_hmis/report_api.rb]:
```suggestion
#   "path"=>"https://BASE.oktapreview.com/oauth2/TOKEN/v1/token",
```

File: drivers/hmis/app/models/ac_hmis/report_api.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# To connect to the API, you need a remote credential for this endpoint. Replace
+# the empty strings below with values from the documentation.
+#
+# creds = GrdaWarehouse::RemoteCredentials::Oauth.where(slug: 'ac_hmis_link').first_or_initialize
```
Comment (gigxz) [inline drivers/hmis/app/models/ac_hmis/report_api.rb]:
```suggestion
# creds = GrdaWarehouse::RemoteCredentials::Oauth.where(slug: 'ac_reports').first_or_initialize
```

**Review by gigxz — APPROVED** [review]
Looks good!

**Pull request merged**